engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/LiteralBranchNode.java

```diff
@@ -0,0 +1,52 @@
+package org.enso.interpreter.node.controlflow.caseexpr;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.node.expression.builtin.text.util.ToJavaStringNode;
+import org.enso.interpreter.runtime.data.text.Text;
+
+@NodeInfo(shortName = "LiteralMatch", description = "Allows matching on literals")
+public abstract class LiteralBranchNode extends BranchNode {
+  private final Object literal;
+
+  private final ConditionProfile textProfile = ConditionProfile.createCountingProfile();
+  private final ConditionProfile numProfile = ConditionProfile.createCountingProfile();
+
+  LiteralBranchNode(Object literal, RootCallTarget branch) {
+    super(branch);
+    this.literal = literal;
+  }
+
+  public static LiteralBranchNode build(Object literal, RootCallTarget branch) {
+    return LiteralBranchNodeGen.create(literal, branch);
+  }
+
+  @Specialization
+  void doText(
+      VirtualFrame frame,
+      Object state,
+      Text target,
+      @Cached("build()") ToJavaStringNode toJavaStringNode) {
+    if (textProfile.profile(literal.equals(toJavaStringNode.execute(target))))
```

Comment:
Converting `Text` to `String` just to do `equals`? I guess we could benefit from a switch to [TruffleStrings](https://github.com/oracle/graal/blob/master/truffle/docs/TruffleStrings.md) then and use [EqualsNode](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/strings/TruffleString.EqualNode.html) which would compare the values without converting to `String`.

Reply:
Yeah, this is the only ugly part in this PR, I admit. 
Thanks for the suggestion, I wasn't entirely sure what is _the proper way_ to deal with this.

Reply:
On the other hand, we use similar technique with `ExpectStringNode`, `ExpectTextNode` (and `ToJavaStringNode`) in various places.

Reply:
So I think that with a comparison of normalized strings this is no longer an option.